### PR TITLE
Add telemetry around report generation in Search.GenerateAuxiliaryData

### PR DESCRIPTION
**Pull request by joelverhagen**
Address https://github.com/NuGet/Engineering/issues/2327

### CODE REVIEW

File: src/Search.GenerateAuxiliaryData/Job.cs [hunk 1]
```diff
@@ -94,15 +98,19 @@ public override async Task Run()
 
             foreach (Exporter exporter in _exportersToRun)
             {
+                var exporterName = exporter.GetType().Name;
+                var reportName = exporter.Name;
+                var stopwatch = Stopwatch.StartNew();
                 try
                 {
                     await exporter.ExportAsync();
+                    TelemetryService.TrackExporterDuration(exporterName, reportName, stopwatch.Elapsed, success: true);
```
Comment (scottbommarito) [inline src/Search.GenerateAuxiliaryData/Job.cs]: You could put the `TrackExporterDuration` call in a `finally`.

Comment (joelverhagen) [reply src/Search.GenerateAuxiliaryData/Job.cs]: I would still need to set `success` 🤷‍♂️

Comment (scottbommarito) [reply src/Search.GenerateAuxiliaryData/Job.cs]: IMO it'd still be cleaner. It's less code and there will never be a discrepancy between the exporterName/reportName in the success and failure cases.

```
var success = false;
try
{
    await exporter.ExportAsync();
    success = true;
}
catch (Exception e)
{
    ...
}
finally
{
    TelemetryService.TrackExporterDuration(exporterName, reportName, stopwatch.Elapsed, success);
}
```

Not a big deal though, of course.

Comment (joelverhagen) [reply src/Search.GenerateAuxiliaryData/Job.cs]: Okay, I can do that

**Pull request merged**